### DISCUSSION TITLE
Row.ToJson does not convert call table fields in Aftersave

### DISCUSSION THREAD

**Discussion by reach2rv** [serenity-is/Serenity#5583] [Help] [answered]
Hello everyone,

I am trying to convert Row to json using Row.ToJson in aftersave method on repository, but I get only few table fields with value (I am not talking about Foreign fields)
most importantly the row id and date field is null  (both Row and Request.Entity has values assigned) but after conversion those values are null).  this doesn't happen while creating new entity but in update I face this issue. 

if anyone has idea how to solve this.. will be really helpful.

**Comment by VictorTomaili** [comment, marked as answer]
When you update a row, it's just collect changed data and create a query for just changed properties in this row object. It's mean you need select this record again for get all updated values.

**Reply by reach2rv** [reply]
is it something like selecting whole entity again? if not can you please give some hint.

**Reply by reach2rv** [reply]
I am lil confuse as if I convert values to Json in  
```C#
public SaveResponse Update(IUnitOfWork uow, SaveRequest<MyRow> request)
{
}
```


then it works but not in aftersave

**Reply by VictorTomaili** [reply]
Yep, best way to handle it, selecting whole row from database again. Your updated data can be changed in database because of triggers, stored procedures etc. Selecting whole row from database should contains latest state of data and brings to you data consistency. 

But if you didn't want to select it from database and you ensure about your data is not changed from database, you can clone OldRow, collect changed fields from Row and set this fields to cloned OldRow.  You can check changed fields with get table fields `this.Row.GetTableFields()` and check with `this.Row.IsFieldChanged()`
